kubelet test is flaky

Appears to deadlock or otherwise hang after a few seconds and then hit the 5-minute timeout.

From start of log:
```
W0406 21:21:50.436464   21680 docker.go:482] found a container with the "k8s" prefix, but too few fields (2): "k8s_unidentified"
...
E0406 21:21:53.397806   21680 server.go:614] Timed out waiting for client to create streams
panic: test timed out after 5m0s
```
followed by stack traces of zillions of goroutines that mostly report being blocked for 4+ minutes.

Recent instances:
https://travis-ci.org/GoogleCloudPlatform/kubernetes/jobs/57351393
https://travis-ci.org/GoogleCloudPlatform/kubernetes/jobs/57389027

cc @dchen1107

--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -162,7 +162,7 @@
 		},
 		{
 			"ImportPath": "github.com/docker/spdystream",
-			"Rev": "e731c8f9f19ffd7e51a469a2de1580c1dfbb4fae"
+			"Rev": "99515db39d3dad9607e0293f18152f3d59da76dc"
 		},
 		{
 			"ImportPath": "github.com/elazarl/go-bindata-assetfs",

--- a/Godeps/_workspace/src/github.com/docker/spdystream/connection.go
+++ b/Godeps/_workspace/src/github.com/docker/spdystream/connection.go
@@ -89,10 +89,25 @@ Loop:
 			if timer != nil {
 				timer.Stop()
 			}
+
+			// Start a goroutine to drain resetChan. This is needed because we've seen
+			// some unit tests with large numbers of goroutines get into a situation
+			// where resetChan fills up, at least 1 call to Write() is still trying to
+			// send to resetChan, the connection gets closed, and this case statement
+			// attempts to grab the write lock that Write() already has, causing a
+			// deadlock.
+			//
+			// See https://github.com/docker/spdystream/issues/49 for more details.
+			go func() {
+				for _ = range resetChan {
+				}
+			}()
+
 			i.writeLock.Lock()
 			close(resetChan)
 			i.resetChan = nil
 			i.writeLock.Unlock()
+
 			break Loop
 		}
 	}

